lib/chef/resource/screensaver.rb

```diff
@@ -0,0 +1,175 @@
+#
+# Author:: Nate Walck (<n8@uber.com>)
+# Copyright:: 2019, Uber, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require_relative "../resource"
+
+class Chef
+  class Resource
+    class Screensaver < Chef::Resource
+      resource_name :screensaver
+      provides(:screensaver) { true }
```

Comment:
this should be the `MacosScreensaver` and `:macos_screensaver` resource that provides `:screensaver` on platform `:mac_os_x` (IIRC).

when you get to the point of code-sharing between multiple different screensaver implementations that need to share the same API you'll need to use a mixin for the common properties, and you'll
wind up needing this:

https://lamont-granquist.gitlab.io/chef/base-modules/